### PR TITLE
Revert "Set MALLOC_TRIM_THRESHOLD_ before interpreter start"

### PR DESCRIPTION
**Pull request by gjoseph92**
Reverts dask/distributed#6681

Closes https://github.com/dask/distributed/issues/6749

cc @pentschev. I know you've found a workaround for dask-cuda in https://github.com/rapidsai/dask-cuda/pull/955, but I'm kind of inclined to revert this anyway and then spend a little time implementing a better solution. Given the unexpected impact this had on dask-cuda, I'm worried it might break things for others too. We always knew the current implementation was a bit hacky anyway.

### CODE REVIEW

**Review by pentschev — APPROVED** [review]
Sounds good to me, thanks @gjoseph92 !

**Comment by crusaderky** [comment]
I'm against this revert. The amount of users benefiting from the automatic MALLOC_TRIM_THRESHOLD_ vastly, vastly outnumbers the number of users that need different env variables in different workers.

**Comment by pentschev** [comment]
> The amount of users benefiting from the automatic MALLOC_TRIM_THRESHOLD_ vastly, vastly outnumbers the number of users that need different env variables in different workers.

Can you provide evidence to sustain this claim? For example, I've never heard of anyone complaining about the deallocation issues that `MALLOC_TRIM_THRESHOLD_` is trying to work around for Dask-CUDA, even though all Dask-CUDA users rely on environment variables that are different among workers. Plus, feel free to correct me if I'm wrong, but if the user simply specifies `MALLOC_TRIM_THRESHOLD_=65535 python` (or whatever process other than `python`), that would have the same effect without any other side effects, am I correct?

Particularly, I dislike the claim that it's fine to keep things broken that we know are broken because it now provides an automatic benefit for some users who could achieve the same behavior by one small change to their workflows (provided my understanding above is indeed correct).

**Comment by crusaderky** [comment]
> For example, I've never heard of anyone complaining about the deallocation issues

This is because, frequently, this is an issue of partial deallocations; in other words it will "solve itself" once the worker has spilled everything to disk - which will cost a 2 order-of-magnitude slowdown.

Also there's the fundamental problem of visibility. In case of OOM (pass the terminate threshold), the user will receive a nebulous KilledWorker, with no immediate indication that the culprit was memory usage (this issue is already discussed elsewhere). Finally, I suspect the that only a fraction of the users suffering from memory problems will spot that there are vast amounts of unmanaged memory, and only the fraction of those will be able to pin the cause to trimming. 

> feel free to correct me if I'm wrong, but if the user simply specifies MALLOC_TRIM_THRESHOLD_=65535 python (or whatever process other than python), that would have the same effect without any other side effects, am I correct? 

Correct. This requires the user to investigate the problem and find the right page on the documentation. Yes we're linking it from the OOM warnings in the log, but that still requires fishing the right lines from the logs, which is a simple activity for you and me but not for someone who just wants to use dask as a tool and has a rightful expectation for it to *just work*. 

TL;DR; *most users are not power users*.

**Comment by pentschev** [comment]
> Correct. This requires the user to investigate the problem and find the right page on the documentation. Yes we're linking it from the OOM warnings in the log, but that still requires fishing the right lines from the logs, which is a simple activity for you and me but not for someone who just wants to use dask as a tool and has a rightful expectation for it to _just work_.
> 
> TL;DR; _most users are not power users_.

I agree that this can be painful for users, and to be clear we're not saying we shouldn't ever do it. Our only ask is that we do so in a manner that will not be problematic for existing use cases, so reverting now before the release would buy us time to work on a solution that would work for everyone. IOW, I still hold my opinion that we shouldn't be ok with changes that break known existing use cases in favor usability changes, we should combine both, and given this is a new feature I would argue in favor of keeping what used to work still working until we come up with a solution to add the new feature that won't break previously functional use cases.

**Review by jrbourbeau — COMMENTED** [review]
@gjoseph92 could you fix the linting issue here?

**Comment by jakirkham** [comment]
The lint appears to be unrelated to the files touched

```
distributed/shuffle/shuffle_extension.py:534: error: Unused "type: ignore" comment
```

**Comment by jakirkham** [comment]
Submitted PR ( https://github.com/dask/distributed/pull/6779 ) to fix it. Should simplify reverting the revert later 🙂

**Comment by jakirkham** [comment]
Re-running the lint now that PR ( https://github.com/dask/distributed/pull/6779 ) is merged.

**Review by jrbourbeau — APPROVED** [review]
Based on the conversation here an in the original issue (https://github.com/dask/distributed/issues/6749) I'm going to merge this PR in order to get `dask-cuda` working again. This is just a temporary revert and, as @gjoseph92 mentioned (https://github.com/dask/distributed/issues/6749#issuecomment-1192557797), we should restore a version of https://github.com/dask/distributed/pull/6681 that addresses the concerns that have been brought up here. Totally acknowledge that there's not an ideal solution here and reverting is frustrating, but there seems to be broader consensus around merging this PR and then quickly getting https://github.com/dask/distributed/pull/6681 restored. 

> Edit: Looks like the merge ref is out-of-date on CI. So this isn't going to work without closing/reopening. Idk if it is worth that level of testing here (given that will restart all CI jobs).

Yeah, agreed we don't need to worry about the linting build here given the changes in https://github.com/dask/distributed/pull/6779

**Comment by jrbourbeau** [comment]
Opened up https://github.com/dask/distributed/issues/6780 so we don't loose track of getting https://github.com/dask/distributed/pull/6681 back into `main`
